AstraEngine.Core/Component.cs

```diff
@@ -30,4 +32,12 @@ public virtual void Tick(double delta)
     {
 
     }
+
+    /// <summary>
+    /// Executed when this component exits the game
+    /// </summary>
+    public virtual void Exit()

```

Comment:
Can you provide an example use case for this? When would exit be called? Who is the caller?